app/src/README-eurotoken-payment-request.md

```diff
@@ -0,0 +1,107 @@
+3. After exchange setup, go to run_coin.py and change GATEWAY_HOSTNAME to the IP address of your computer running the python server
+4. Now make some changes to TrustChain App. Basically change my IP (192.168.0.106) to your computer IP:
+5. go to app/src/main/res/xml/network_security_config.xml and change 192.168.0.106 to your local IP
+6. go to common/build.gradle and change ONLY IP of DEFAULT_GATEWAY_HOST to your IP
+7. Run backend exchange with python3 run_coin.py
+
+## Challenges:
+Working with this project came along with some challenges, namely: 
+
+- We could not fully test scenario 3 due to not being able to acquire the live API keys from Robert
+- The superapp code is not tested, hence there could be hidden bugs that might have contributed to difficulties of the project
+- There was no documentation with the code we worked with. During the progress, we did not know how to use specific code and has to go throug all the code with sometimes having to debug that code to understand it.
+- By default, ConfiApp used QR code as E2E communication, which enable the application to know the receiver of the payment - the receiver is known. However, with our payment request implementation, we can create and share link but there is actually no way to know who is the receiver for sure (IP address, Port, Identity, etc).  
+- When you doing a project that is outsourced, you need coding conventions. With the superapp code there were no coding conventions. We did not know where to put our codes and what are the structure. Coding style were also inconsistent throughout the project.
+
+
+## Suggestions:
+- The link needs to be set up with a website which display payment information and methods so that the payer who does not have the EuroToken app can still make use of the EuroToken payment request system.
+
+- Even though our implementation of link sharing is very straight forward, during this progress there is no guarantee that the person behind the public key is the intended benficiary. The suggestion for the future is that, a validation mechanism should be implemented either in the server (with a trusted authority) or user to make sure the link is shared by a receiver linked to the strong identity (for example passport-enrolled identity in the app)
+
+
+## Testings:
+
+Tests were not written for Valuetransfer app previously, however we tested our code. There are two types of test you can find in the repository:
+Unit Tests:
+
+- Handling Incoming links
+- Correct Amount input
+- IBAN validation
+
+Integration Tests:
+- Link Signature validation
+
+
+We had difficulties with setting E2E intergration tests since you will need to setup an instance of IPv8. We have also tried to do mock testing, however we came across a variety of problems. 
+
+
+
+## Support:
+If you come across issues with this feature or simply have implementation questions, you can send them to:
+t.l.nguyen@student.tudeflt.nl
```

Comment:
Your e-mail address wil be invalidated after you graduate. I would leave out this section :)